docs/features/scripting.rst

```diff
@@ -96,29 +118,405 @@ where:
+
+As a baseline lets start off with the familiar FirstApp
+::
+
+    #!/usr/bin/env python
+    """A simple cmd2 application."""
+    import cmd2
+
+
+    class FirstApp(cmd2.Cmd):
+        """A simple cmd2 application."""
+        def __init__(self):
+          shortcuts = cmd2.DEFAULT_SHORTCUTS
+          shortcuts.update({'&': 'speak'})
+          super().__init__(shortcuts=shortcuts)
+
+          # Make maxrepeats settable at runtime
+          self.maxrepeats = 3
+          self.add_settable(cmd2.Settable('maxrepeats', int, 'max repetitions for speak command', self))
+
+
+        speak_parser = cmd2.Cmd2ArgumentParser()
+        speak_parser.add_argument('-p', '--piglatin', action='store_true', help='atinLay')
+        speak_parser.add_argument('-s', '--shout', action='store_true', help='N00B EMULATION MODE')
+        speak_parser.add_argument('-r', '--repeat', type=int, help='output [n] times')
+        speak_parser.add_argument('words', nargs='+', help='words to say')
+
+        @cmd2.with_argparser(speak_parser)
+        def do_speak(self, args):
+            """Repeats what you tell me to."""
+            words = []
+            for word in args.words:
+                if args.piglatin:
+                    word = '%s%say' % (word[1:], word[0])
+                if args.shout:
+                    word = word.upper()
+                words.append(word)
+            repetitions = args.repeat or 1
+            for _ in range(min(repetitions, self.maxrepeats)):
+                # .poutput handles newlines, and accommodates output redirection too
+                self.poutput(' '.join(words))
+
+    if __name__ == '__main__':
+        import sys
+        c = FirstApp()
+        sys.exit(c.cmdloop())
+
+
+Lets start off on the wrong foot::
+
+  app('speak'
+  print('Working')
+
+::
+
+  SyntaxError: unexpected EOF while parsing
+  (Cmd) run_pyscript script.py
+    File "<string>", line 2
+      app('speak'
+                ^
+  SyntaxError: unexpected EOF while parsing
+
+cmd2 pyscripts require **valid** python code as a first step. 
+
+.. warning::
+
+  It is a common misconception that all application exceptions will "bubble" up from below. Unfortunately or fortunately
+  this is not the case. CMD2 sinkholes all application exceptions and there are no means to handle them.
+
+
+When executing the ``speak`` command without parameters you see the following error::
+
+  (Cmd) speak 
+  Usage: speak [-h] [-p] [-s] [-r REPEAT] words [...]
+  Error: the following arguments are required: words
+
+Even though this is a fully qualified CMD2 error the py_script must look for this error and perform error checking.::
+
+  app('speak')
+  print("Working")
+
+::
+
+  (Cmd) run_pyscript script.py 
+  Working
+  (Cmd) 
+
+You should notice that no error message is printed. Let's utilize the ``CommandResult`` 
+object to inspect the actual returned data.::
+
+  result = app('speak')
+  print(result)
+::
```

Comment:
Need to fix doc build warning:
```
Warning, treated as error:
/home/runner/work/cmd2/cmd2/docs/features/scripting.rst:275:Literal block ends without a blank line; unexpected unindent.
```

Reply:
is there an rst linter?? I was using vscode RST extension but didn't see anything wrong.